### PR TITLE
PR: Add ciocheck linter formater tester

### LINKED ISSUE #3535: Run ciocheck as part of our tests to test code style
https://github.com/ContinuumIO/ciocheck

Package available at: https://anaconda.org/continuumcrew/ciocheck

`conda install ciocheck -c conda-forge -c continuumcrew`

Moving to pypi and conda-forge soon.

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #3535

### CODE REVIEW

**Comment by goanpeca** [comment]
This is ready for travis at the moment and only when using conda, cause pypi is now super stupid and I have to make a new release cause I made a mistake on upload and now you cant reupload a file... so stupid.

**Comment by ccordoba12** [comment]
> I made a mistake on upload and now you cant reupload a file... so stupid.

Yeah, PyPi doesn't have build numbers :-)

> this should be enough anyway to get us going

Nice!! Let me release 3.0.2 first and I'll come back right away to this one.

**Comment by goanpeca** [comment]
@ccordoba12 please activate circleci builds

File: .travis.yml [hunk 1]
```diff
@@ -39,10 +39,14 @@ install:
...
   - if [ "${USE_CONDA}" = "true" ]; then
         source activate test-environment;
+        conda install ciocheck -c conda-forge -c continuumcrew;
     fi
   - pip install coveralls
 
 script:
+  - if [ "${USE_CONDA}" = "true" ]; then
+        ciocheck spyder;
+    fi
```
Comment (ccordoba12) [inline .travis.yml]: Please remove these changes. Let's leave ciocheck to work in CircleCI only.

**Comment by ccordoba12** [comment]
After you undo the `travis.yml` changes, I'll merge this one and then #3298.

**Comment by ccordoba12** [comment]
@goanpeca, I did the necessary changes myself :-)

**Pull request merged**